### DISCUSSION TITLE
Azure Monitor Custom Metrics

### DISCUSSION THREAD

**Discussion by FreddyAyala** [tomkerkhove/promitor#1284] [Q&A] [answered]
As a requirement of our client we should be able to scrape custom metrics from Azure Monitor in order to send metrics from Azure to Prometheus that is not supported natively, for example metrics about Security Center and Azure Policy, like in this example using the api:

https://francecentral.monitoring.azure.com/<ResourceId>/metrics

{ 
    "time": "2020-09-16T08:03:20", 
    "data": { 
        "baseData": { 
            "metric": "CustomMetric1", 
            "namespace": "CustomVmMetrics", 
            "dimNames": [ 
              "QueueName", 
              "MessageType" 
            ], 
            "series": [ 
              { 
                "dimValues": [ 
                  "ImagesToProcess", 
                  "JPEG" 
                ], 
                "min": 3, 
                "max": 20, 
                "sum": 28, 
                "count": 3 
              } 
            ] 
        } 
    } 
} 

Is it possible to scape cutom metrics using Promitor?

**Comment by tomkerkhove** [comment, marked as answer]
We [support a ton of metric scrapers](https://promitor.io/configuration/v2.x/metrics/#supported-azure-services) which make it convenient to scrape a given Azure Monitor metric.

However, we don't support every service yet and have a lot of them on the backlog but you can still use them if you [use our generic scraper](https://promitor.io/configuration/v2.x/metrics/generic-azure-resource) which allows you to scrape any of the metrics in [the official list of Azure Monitor metrics](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported).

You are talking about [this API](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-custom-overview) I presume? This is planned in https://github.com/tomkerkhove/promitor/issues/80 but not supported now. How soon would you need this? Taking into account this is an OSS project.
